lib/expense_tracker/expenses/expense_server.ex

```diff
@@ -0,0 +1,111 @@
+defmodule ExpenseTracker.Expenses.ExpenseServer do
+  @moduledoc """
+  This module provides a GenServer for managing user's expenses.
+  """
+
+  import Ecto.UUID, only: [generate: 0]
+
+  alias ExpenseTrackerWeb.Endpoint
+
+  use GenServer
+
+  # Public API
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, %{expenses: %{}}, name: __MODULE__)
+  end
+
+  def add_expense(user_id, expense) do
+    GenServer.call(__MODULE__, {:add_expense, user_id, expense})
+  end
+
+  def get_expenses(user_id) do
+    GenServer.call(__MODULE__, {:get_expenses, user_id})
+  end
+
+  def get_expense_by_id(user_id, expense_id) do
+    GenServer.call(__MODULE__, {:get_expense_by_id, user_id, expense_id})
+  end
+
+  def get_total_by_user(user_id) do
+    GenServer.call(__MODULE__, {:get_total_by_user, user_id})
+  end
+
+  def delete_expense(user_id, expense_id) do
+    GenServer.call(__MODULE__, {:delete_expense, user_id, expense_id})
+  end
+
+  def clean do
+    GenServer.call(__MODULE__, :clean)
+  end
+
+  # Server Callbacks
+
+  def init(state) do
+    {:ok, state}
+  end
+
+  def handle_call({:add_expense, user_id, expense}, _from, %{expenses: expenses} = state) do
+    expense = Map.put(expense, :id, generate())
+
+    expense =
+      case Map.get(expense, :date) do
+        nil -> Map.put(expense, :date, Date.utc_today())
+        _ -> expense
+      end
+
+    updated_expenses =
+      expenses
+      |> Map.update(user_id, [expense], fn existing -> [expense | existing] end)
+
+    Endpoint.broadcast("expenses_#{user_id}", "expense_created", expense)
+
+    new_state = %{state | expenses: updated_expenses}
```

Comment:
```suggestion
    new_state = %{state | expenses: updated_expenses}
    Endpoint.broadcast("expenses_#{user_id}", "expense_created", expense)
```
It seems better if we update the state first and the broadcast. With the current implementation, if the broadcast fails, the state will still have the new expense, leading to inconsistency.